src/common/snippets/src/lowered/pass/insert_tail_loop.cpp

```diff
@@ -8,47 +8,230 @@
 #include "snippets/lowered/loop_manager.hpp"
 #include "snippets/lowered/pass/init_loops.hpp"
 #include "snippets/snippets_isa.hpp"
+#include "snippets/utils.hpp"
 #include "snippets/itt.hpp"
 
 namespace ov {
 namespace snippets {
 namespace lowered {
 namespace pass {
+void InsertTailLoop::propagate_updated_subtensor_through_loop(const LinearIR& linear_ir,
+                                                              const LinearIR::LoopManager::LoopInfoPtr& loop_info,
+                                                              LinearIR::container::const_iterator begin,
+                                                              LinearIR::container::const_iterator end,
+                                                              const size_t new_dim_value) {
+    std::map<lowered::PortDescriptorPtr, snippets::VectorDims> original_shapes;
+    // First step: set new dim value to the corresponding entry_points' dimensions
+    if (new_dim_value != SIZE_MAX) {
+        for (const auto& port : loop_info->entry_points) {
+            if (port.is_incremented) {
+                const auto& expr = port.expr_port->get_expr();
+                const auto node = expr->get_node();
+                auto desc = port.expr_port->get_descriptor_ptr();
+                auto subtensor = desc->get_subtensor();
+                if (port.dim_idx < subtensor.size()) {
+                    *(subtensor.rbegin() + port.dim_idx) = new_dim_value;
```

Comment:
Actually the current approach implemented in `propagate_updated_subtensor_through_loop` don't work for all possible cases: it covers only the current subgraphs which can be tokenized.
Moreover, during the work I realized that the approach based on propagating subtensors is generally quite poorly suited for updating Matmul blocking parameters: it is expensive (since we are forced to traverse the entire loop body and call the shape inference for each node), and hard to implement (especially to support a common case).

Approach based on specific iteration handler (ticket 119851) seems much more appropriate. So I would rather concentrated on that task.

However, since the review of this PR was already partially passed, I suggest leaving the `propagate_updated_subtensor_through_loop` as is.